src/middleware/error.middleware.js

```diff
@@ -0,0 +1,16 @@
+import { ApiError } from '../exceptions/api.error.js';
+
+export const errorMiddleware = (error, req, res, next) => {
+  if (error instanceof ApiError) {
+    res.status(error.status).send({
+      message: error.message,
+      errors: error.errors,
+    });
+  }
+
+  res.statusCode = 500;
```

Comment:
The line setting `res.statusCode = 500;` should be inside an `else` block to ensure it only executes when the error is not an instance of `ApiError`. Otherwise, it will override the status code set in the `if` block.